bpfd/src/multiprog/tc.rs

```diff
@@ -210,7 +210,7 @@ impl TcDispatcher {
                     bpf.set_global(name, value.as_slice(), true);
                 }
 
-                let (_, map_pin_path) = calc_map_pin_path(**k, v.data.map_owner_id());
+                let (_, map_pin_path) = calc_map_pin_path(k.to_owned(), v.data.map_owner_id());
```

Comment:
Yeah it could but unfortunately with the design of all the map stuff right now that change would have far-reaching effects for this PR. 

Let me tackle this in another one 👍